app/models/consumer.rb

```
@@ -33,6 +33,13 @@ def api_ids
     apis.map(&:id)
   end
 
+  def promote_to_prod(api_ref)
+    api = ApiRef.find_by(name: api_ref).api
```

What happens if a record isn't found here? Should we raise an exception? Return false immediately after this line?

---

in it's current state I don't think it can receive an invalid ApiRef name as it loops through the consumer's apis and pulls the `api_ref[:name]` from each. These are the values that get sent in if the consumer has access to the API in sandbox, and the ref is provided via the request. I don't believe there is a way for an invalid ApiRef name to be provided to the array we loop through, but I've been surprised before.

From a safety perspective for any potential future use, yeah, it makes sense to put some safety catch here. I'd bet we could add these checks in a few spots as well.

---

[45b1376](https://github.com/department-of-veterans-affairs/lighthouse-platform-backend/pull/78/commits/45b1376fb49e8355f44bdef73be0c4ebf05b1bc5)